tests/basic/storage_pool_test.go

```diff
@@ -9597,16 +9601,18 @@ func scheduleApps() []*scheduler.Context {
 	contexts := make([]*scheduler.Context, 0)
 	for i := 0; i < Inst().GlobalScaleFactor; i++ {
 		log.Infof("Deploy app %v", i)
-		contexts = append(contexts, ScheduleApplications(fmt.Sprintf("pooltest-%d", i))...)
+		contexts = append(contexts, ScheduleApplicationsOnNamespace(
+			testingAppNamespace, fmt.Sprintf("pooltest-%d", i))...)
 	}
 	ValidateApplications(contexts)
 	return contexts
 }
 
-func pickPoolToResize(contexts []*scheduler.Context) string {
-	poolIDToResize, err := GetPoolIDWithIOs(contexts)
+func pickPoolToResize() string {
+	poolIDsInUseByTestingApp, err := GetPoolsInUse(TESTING_STORAGE_CLASS)
```

Comment:
Our previous approach of locating a volume with ongoing I/O, although more generic, added an uncontrollable element, making failures indeterministic and harder to reproduce or debug. In fact, the issue was observed from "passing the fio app required using the app-list", roughly ~5-15% test failures were observed due to this particular failure. 
For system tests or QA tests, I believe your suggestion about genericity is valid. However, in the context of functional tests within our CI/CD pipeline, the purpose is to verify that a given feature or function of the application behaves as expected under controlled circumstances. We aim to create a reliable and deterministic test environment to to ensure consistent and predictable results, which would allows us to quickly and accurately identify regressions or issues introduced by new code, that directly aids developers in identifying and fixing issues promptly. 

The decision to enhance our test stability and reliability was made internally, Eventually we would want to encode one testing app in the test code instead of being passed from jenkins. I'll be happy to share more of the rationals over a call if needed, please feel free to DM me on Slack.